[Kee Song Yang] iP

s0ngyang opened the pull request:
# Nexus 
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Nexus is
- text-based
- easy to learn
- ~~FAST~~ ***SUPER FAST*** to use

All you need to do is,

1. download it from [here](https://github.com/s0ngyang/ip/releases).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you're a Java programmer, you may use it to practise Java too! Here's how to execute the `main` method:
```
public static void main(String[] args) {
    String path = String.join(File.separator, "src", "main", "data", "nexus.txt");
    new Nexus(path).run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

s0ngyang committed:
Renamed Chatbot

s0ngyang committed:
Added chatbot echo

s0ngyang committed:
Added list

s0ngyang committed:
Added Mark as done

s0ngyang committed:
Added different types of tasks

s0ngyang committed:
Added automated testing

s0ngyang committed:
Merge pull request #1 from s0ngyang/week-2

Week 2

s0ngyang committed:
Merge from branch

s0ngyang committed:
Merge branch 'week-2'

s0ngyang committed:
Merge branch 'master' of https://github.com/s0ngyang/ip

s0ngyang committed:
Handled errors minimally

s0ngyang committed:
Added delete

s0ngyang committed:
Changed switch clause to adhere to convention

s0ngyang committed:
Push from work laptop

s0ngyang committed:
Push from desktop

s0ngyang committed:
WIP Save

s0ngyang committed:
Implement Save

s0ngyang committed:
Remove println

s0ngyang committed:
Add datetime to Deadline

s0ngyang committed:
Add datetime to Event

s0ngyang committed:
Ui abstracted

s0ngyang committed:
Add nexus package

s0ngyang committed:
Merge branch 'add-gradle-support'

s0ngyang committed:
Setup gradle

s0ngyang committed:
Add JUnit test for Todo and Deadline

s0ngyang committed:
Add javadoc for half the classes and methods

s0ngyang committed:
Add parseFind header

s0ngyang committed:
WIP parseFind

s0ngyang committed:
Add find command

s0ngyang committed:
Remove wildcard imports

s0ngyang committed:
Merge branch 'branch-A-CodingStandard'

s0ngyang committed:
Merge branch 'branch-Level-9'

lamchenghou commented on src/main/java/nexus/components/Storage.java:
```diff
@@ -0,0 +1,118 @@
+import java.util.Objects;
+import java.util.Scanner;
+
+public class Storage {
```
Perhaps you can include JavaDocs for the more complex public methods in Storage.

lamchenghou commented on src/main/java/nexus/components/Storage.java:
```diff
@@ -0,0 +1,118 @@
+                        break;
+                    case "delete":
+                        break;
+                    }
```
I like that you followed CS2103T's way of writing switch-case statements.

lamchenghou commented on src/main/java/nexus/Nexus.java:
```diff
@@ -0,0 +1,59 @@
+     * @param path String
+     */
+    public Nexus(String path) {
+        this.storage = new Storage(path);
+        this.ui = new Ui();
+        this.list = new TaskList(storage.loadTasks());
```
Should we still use the qualified `this.` name if the field is not shadowed?

lamchenghou commented on src/main/java/nexus/components/Parser.java:
```diff
@@ -0,0 +1,175 @@
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+    private static void parseMark(TaskList list, String[] data, Storage storage, Ui ui) throws InvalidInputException {
```
Nit: Just a small note that you have exceeded the soft limit (110 characters) on some lines.

lamchenghou commented on src/main/java/nexus/task/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
```
Nit: Maybe you can choose a more descriptive name such as `endDateTime`.

lamchenghou commented on src/main/java/nexus/task/Event.java:
```diff
@@ -0,0 +1,42 @@
+    /**
+     * Create an event using description, start date & time and end date & time.
+     *
+     * @param description String
+     * @param from Start date & time
+     * @param to End Date & time
```
I believe there should be punctuation at the end of every parameter description.

lamchenghou commented on src/test/java/nexus/task/TodoTest.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    // think this does not count because it depends on Parser class
+//    @Test
+//    public void testEmptyTodo() {
+//        String path = "src" + File.separator + "test" + File.separator + "data" + File.separator + "nexus.txt";
+//        assertThrows(InvalidInputException.class,
+//                () -> Parser.parseInput(new Ui(), new Storage(path), new TaskList(),"todo"));
+//    }
```
Nit: can indent / remove commented out code, though it looks like it's WIP 😄

s0ngyang replied:
Thank you for taking the time to review my code! I have made changes according to your comments :)

lamchenghou reviewed (commented):
Looks good 👍, overall a well styled codebase with well named variables. I enjoyed reading through your code 😄, and even learned some new methods of String manipulation.

Some styling issues and nits to fix.

s0ngyang committed:
Fix todo spacing bug

s0ngyang committed:
Clean up code according to reviewer comments

s0ngyang committed:
Build jar

s0ngyang committed:
Clean up main and test code according to checkstyle

s0ngyang committed:
Add GUI

s0ngyang committed:
GUI integrated

s0ngyang committed:
Implement GUI

s0ngyang committed:
Add welcome message

s0ngyang committed:
Add assertions for Parser

s0ngyang committed:
Run checkstyleMain

s0ngyang committed:
Remove unused methods

s0ngyang committed:
Merge pull request #3 from s0ngyang/branch-A-CodeQuality

Branch a code quality

s0ngyang committed:
Merge branch 'master' into branch-A-Assertions

s0ngyang committed:
Merge pull request #4 from s0ngyang/branch-A-Assertions

Branch a assertions

s0ngyang committed:
Change user image

s0ngyang committed:
Merge branch 'master' of https://github.com/s0ngyang/ip

s0ngyang committed:
Fix nexusImage typo

s0ngyang committed:
Enable text wrap when text overruns

s0ngyang committed:
Add simplified commands

Each command has a shortened version of itself, except for "list".
Most commands can be executed with the first letter,
except "delete" whose shortform is "del".

s0ngyang committed:
Change user and nexus images

s0ngyang committed:
Add border and padding

s0ngyang committed:
Add README

s0ngyang committed:
Add window name

s0ngyang committed:
Remove goodbye message

Users will click X on the top right to exit the application.

s0ngyang committed:
Update README

s0ngyang committed:
Change filepath

s0ngyang committed:
Update Ui picture

s0ngyang committed:
Handle index out of bounds error for mark, unmark and delete

s0ngyang committed:
Merge branch 'master' of https://github.com/s0ngyang/ip

s0ngyang committed:
Edit error message for index error

s0ngyang committed:
Handle invalid datetime input
